### PR TITLE
Incorporate compiler-plugin code-modifiers to open-api service definition generation

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2472](https://github.com/ballerina-platform/ballerina-standard-library/issues/2472)

### CODE REVIEW

**Comment by lnash94** [comment]
I'm afraid that whether we can move openapi spec generation API [generateOASForGivenFormat](https://github.com/ballerina-platform/openapi-tools/blob/8c9ea0365e3c6334aa57994139bea71bac27b5a1/openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java#L169)  to the new API [`generateOAS`](https://github.com/ballerina-platform/openapi-tools/blob/8c9ea0365e3c6334aa57994139bea71bac27b5a1/openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ServiceToOpenAPIConverterUtils.java#L228)  since we deprecated that `generateOASForGivenFormat` .

**Comment by ayeshLK** [comment]
@lnash94 yes we need to do that. But, since it is not a critical blocker will plan it for a patch release.

File: module-ballerina-openapi/annotation.bal [hunk 1]
```diff
@@ -31,7 +31,7 @@ public type ServiceInformation record {|
     string[]? excludeTags = [];
     string[]? excludeOperations = [];
     boolean failOnErrors = true;
-    boolean embed = true;
```
[the remark this answers is not in the document]
Comment (ayeshLK) [reply module-ballerina-openapi/annotation.bal:34]: This configuration is used to enable or disable the open-api doc auto generation. Auto generation is hightly useful if we could provide a proper REST client which could use the generated doc to shape the client interactions with the server. Currently we are working on a design for a proper REST client which could facilitate with this open-api doc generation. Until it is available we decided to disable this by default.

File: openapi-extension/src/main/java/io/ballerina/openapi/extension/OpenApiCodeModifier.java [hunk 2]
```diff
@@ -17,15 +17,17 @@
...
 import io.ballerina.compiler.syntax.tree.SyntaxKind;
-import io.ballerina.projects.plugins.CodeAnalysisContext;
-import io.ballerina.projects.plugins.CodeAnalyzer;
+import io.ballerina.openapi.extension.doc.OpenApiInfoUpdaterTask;
+import io.ballerina.projects.plugins.CodeModifier;
+import io.ballerina.projects.plugins.CodeModifierContext;
 
 /**
- * {@code OpenApiCodeAnalyzer} handles syntax analysis for HTTP Services.
+ * {@code OpenApiCodeModifier} handles required code-modification related to open-api doc generation for http-services.
```
[the remark this answers is not in the document]
Comment (ayeshLK) [reply openapi-extension/src/main/java/io/ballerina/openapi/extension/OpenApiCodeModifier.java:25]: Fixed with [15516c73202af27455e1acb66cf266b31bf8da1c](https://github.com/ballerina-platform/openapi-tools/pull/955/commits/15516c73202af27455e1acb66cf266b31bf8da1c)

**Review by lnash94 — APPROVED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]
Changes LGTM

**Pull request merged**